security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -129,19 +129,22 @@ func (c *citadelClient) getTLSDialOption() (grpc.DialOption, error) {
 			return nil, fmt.Errorf("failed to append certificates")
 		}
 	}
-
-	config := tls.Config{}
-	config.RootCAs = certPool
-
-	if ProvCert != "" {
-		// Load the certificate from disk
-		certificate, err := tls.LoadX509KeyPair(ProvCert+"/cert-chain.pem", ProvCert+"/key.pem")
-		if err != nil {
-			return nil, fmt.Errorf("cannot load key pair: %s", err)
-		}
-
-		config.Certificates = []tls.Certificate{certificate}
+	var certificate tls.Certificate
+	config := tls.Config{
+		Certificates: []tls.Certificate{certificate},
+		GetClientCertificate: func(*tls.CertificateRequestInfo) (*tls.Certificate, error) {
```

Comment:
Do we need to do anything to recreate the connection? does it have 30min max connection age or something to make it reconnect?

Reply:
we do not need to recreate the connection, as every time there is connection failure it will try to reconnect.